webpack/assets/javascripts/react_app/components/users/PersonalAccessTokens/PersonalAccessTokenForm.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import * as Yup from 'yup';
+import { Button } from 'patternfly-react';
+import UUID from 'uuid/v1';
+import ForemanForm from '../../common/forms/ForemanForm';
+import TextField from '../../common/forms/TextField';
+import ForemanModal from '../../ForemanModal';
+import { useForemanModal } from '../../ForemanModal/ForemanModalHooks';
+import { maxLengthMsg, requiredMsg } from '../../common/forms/validators';
+import { translate as __ } from '../../../common/I18n';
+import { submitForm } from '../../../redux/actions/common/forms';
+import DateTime from '../../common/forms/DateTime/DateTime';
+
+const MODAL_ID = 'personal-access-tokens-form-modal';
+
+const tokenFormSchema = Yup.object().shape({
+  name: Yup.string()
+    .max(...maxLengthMsg(254))
+    .required(requiredMsg()),
+  expires_at: Yup.date().min(new Date(), __('Cannot be in the past')),
+});
+
+const PersonalAccessTokenForm = ({ controller, url, initialValues }) => {
+  const dispatch = useDispatch();
+  const { setModalOpen, setModalClosed } = useForemanModal({
+    id: MODAL_ID,
+  });
+
+  const handleSubmit = async (values, actions) => {
+    await dispatch(
+      submitForm({
+        url,
+        values: { ...values, controller },
+        item: 'personal_access_token',
+        message: __('Personal Access Token was successfully created.'),
+      })
+    );
+    setModalClosed();
+  };
+
+  return (
+    <p>
+      <Button bsStyle="success" className="btn-lg" onClick={setModalOpen}>
+        {__('Add Personal Access Token')}
+      </Button>
+
+      <ForemanModal id={MODAL_ID} title={__('Create Personal Access Token')}>
+        <ForemanModal.Header />
+        <ForemanForm
+          onSubmit={(values, actions) => handleSubmit(values, actions)}
+          initialValues={initialValues}
+          validationSchema={tokenFormSchema}
+          onCancel={setModalClosed}
+        >
+          <TextField
+            name="name"
+            type="text"
+            label={__('Name')}
+            inputClassName="col-md-6"
+            required
+          />
+          <DateTime
+            id={UUID()}
+            label={__('Expires')}
+            isRequired={false}
+            inputClassName="col-md-6"
+            placement="bottom"
+            inputProps={{ name: 'expires_at' }}
+            hideValue
+          />
```

Comment:
**OPTIONAL** The `DateTime` component is not really used and obsolate, we should use [FormField](https://theforeman.github.io/foreman/?path=/story/components-form--form-field)
```suggestion
          <FormField
            type="dateTime"
            id={UUID()}
            label={__('Expires')}
            isRequired={false}
            inputClassName="col-md-6"
            placement="bottom"
            inputProps={{ name: 'expires_at' }}
            hideValue
          />
```
Although I take as my personal failure, that this way doesn't have much of a documentation and is still not as easy to use with `Formik`, so I'm OK with merging this as is and I'll update in follow-up PR.